models/intermediate/int_intercom__conversation_part_events.sql

```diff
@@ -29,6 +29,17 @@ conversation_contact_events as (
 
 ), 
 
+--Obtains the first and last values for conversations where the part type was authored by a team
+conversation_team_events as (
+  select
+    conversation_id,
+    first_value(author_id ignore nulls) over (partition by conversation_id order by created_at asc, conversation_id rows unbounded preceding) as first_team_id,
+    first_value(author_id ignore nulls) over (partition by conversation_id order by created_at desc, conversation_id rows unbounded preceding) as last_team_id
+  from conversation_part_history
+
+  where author_type = 'team'
```

Comment:
Should we be using `author_id` and `author_type` for this cte? I tested a few team changes with the seed data and noticed they were not being reflected in the output model because we were looking at author fields. I also don't think there would be an instance where a team would be the author of a conversation part, but I could see the team being assigned.

I tested on my end and believe if we update these to check the first value of `assigned_to_id` and our where clause to reference the 'team' type for `assigned_to_type` then we will see accurate results. Would you be able to confirm on your end as well?
```suggestion
    first_value(assigned_to_id ignore nulls) over (partition by conversation_id order by created_at asc, conversation_id rows unbounded preceding) as first_team_id,
    first_value(assigned_to_id ignore nulls) over (partition by conversation_id order by created_at desc, conversation_id rows unbounded preceding) as last_team_id
  from conversation_part_history

  where assigned_to_type = 'team'
```